src/main/scala/sample/reactivekafka/KafkaReaderCoordinator.scala

```diff
@@ -41,19 +41,53 @@ class KafkaReaderCoordinator(mat: Materializer, topicName: String) extends Actor
       .kafkaOffsetsStorage()
       .commitInterval(1200 milliseconds)
     )
+
     log.debug("Starting the reader")
-    Source(consumerWithOffsetSink.publisher)
-      .map(processMessage)
+    val currencyRateSource = Source(consumerWithOffsetSink.publisher)
+    val currencyBroadcaster = Source(currencyRateSource.runWith(Sink.publisher(true)))
+
+    val alertsFlow = Flow[MessageAndMetadata[Array[Byte], CurrencyRateUpdated]]
+      .map(processAlert)
+      .filter(alert => alert != None)
+      .map(alert => alert.get)
+
+
+    // commit offset so we don't process the kafka message again
+    currencyBroadcaster
       .withAttributes(ActorAttributes.supervisionStrategy(processingDecider))
-      .to(consumerWithOffsetSink.offsetCommitSink).run()
```

Comment:
I understand your flow as:
1. Read from the source (currencyBroadcaster)
2. Write to commit sink 
3. Do more stuff with the msg (in other streams)

If we resigned from manual commit, the result would be actual the same:
1. Read from the source (`currencyBroadcaster`)
2. (Automatically commit) <- this happens automatically under the hood in Kafka Client
3. Do more stuff with the msg

As far as I understand, the "at least once delivery" semantics gives us possibility to make sure that we can put our message through some internal processing and then commit it as "delivered". Without any internal processing between reading and committing, it doesn't give us much.